### DISCUSSION TITLE
Extracting data from exoudsiie

### DISCUSSION THREAD

**Discussion by burritotukituki** [idaholab/moose#27508] [Q&A General] [answered]
Hi experts,

I'm not entirely sure if this is the appropriate platform for my query, but I could really use some guidance.

I'm currently running numerous small simulations, each of which generates an exodus file. My goal is to extract a specific variable (cell data) but only from a particular block within these files.

I've been attempting to achieve this using meshio in Python, but I've encountered difficulties in correlating the blocks with my data. It seems that I cannot access the blocks when I open the file with meshio.

Could someone please suggest an approach for dealing with this data extraction issue?

Thanks in advance!

**Comment by GiudGiud** [comment]
Hello

I think you can work with the vtk python package to do this. But I have never done it. 
Alternatively, Paraview has programmable filters that may be friendlier than python packages

Guillaume

**Comment by GregVernon** [comment, marked as answer]
The best approach is to build the [SEACAS project](https://github.com/sandialabs/seacas) ([documentation](https://sandialabs.github.io/seacas-docs/sphinx/html/index.html#)), which is the project that "owns" [Exodus](https://github.com/sandialabs/seacas?tab=readme-ov-file#exodus) (SEACAS might already be built by MOOSE?) and use one of the following approaches:

1. `exodus.py` ([documentation](https://sandialabs.github.io/seacas-docs/sphinx/html/exodus.html))
2. `exo2mat` / `mat2exo` ([documentation](https://sandialabs.github.io/seacas-docs/mat2exo.pdf))
3. `exotxt` / `txtexo` ([documentation](https://sandialabs.github.io/seacas-docs/sphinx/html/index.html#exotxt))

**Reply by cticenhour** [reply]
We do package the SEACAS tools for our users. To install in your MOOSE conda environment, simply:
```
conda install moose-seacas
```

**Comment by burritotukituki** [comment]
Hi all,
thanks for the great ideas. I exported the results as VTK and used meshio to read the files. That worked, thanks!!
